website/www/site/content/en/blog/deploy-python-pipeline-on-flink-runner.md

```diff
@@ -0,0 +1,857 @@
+                topic=self.topic,
+                expansion_service=self.expansion_service,
+            )
+        )
+
+
+def run(argv=None, save_main_session=True):
+    parser = argparse.ArgumentParser(description="Beam pipeline arguments")
+    parser.add_argument(
+        "--deploy",
+        dest="deploy",
+        action="store_true",
+        default="Flag to indicate whether to deploy to a cluster",
+    )
+    parser.add_argument(
+        "--bootstrap_servers",
+        dest="bootstrap",
+        default="host.docker.internal:29092",
+        help="Kafka bootstrap server addresses",
+    )
+    parser.add_argument(
+        "--input_topic",
+        dest="input",
+        default="input-topic",
+        help="Kafka input topic name",
+    )
+    parser.add_argument(
+        "--output_topic",
+        dest="output",
+        default="output-topic-beam",
+        help="Kafka output topic name",
+    )
+    parser.add_argument(
+        "--group_id",
+        dest="group",
+        default="beam-word-len",
+        help="Kafka output group ID",
+    )
+
+    known_args, pipeline_args = parser.parse_known_args(argv)
+
+    print(known_args)
+    print(pipeline_args)
+
+    # We use the save_main_session option because one or more DoFn elements in this
+    # workflow rely on global context. That is, a module imported at the module level.
+    pipeline_options = PipelineOptions(pipeline_args)
+    pipeline_options.view_as(SetupOptions).save_main_session = save_main_session
+
+    expansion_service = None
+    if known_args.deploy is True:
+        expansion_service = get_expansion_service()
+
+    with beam.Pipeline(options=pipeline_options) as p:
+        (
+            p
+            | "ReadWordsFromKafka"
+            >> ReadWordsFromKafka(
+                bootstrap_servers=known_args.bootstrap,
+                topics=[known_args.input],
+                group_id=known_args.group,
+                expansion_service=expansion_service,
+            )
+            | "CalculateAvgWordLen" >> CalculateAvgWordLen()
+            | "WriteWordLenToKafka"
+            >> WriteWordLenToKafka(
+                bootstrap_servers=known_args.bootstrap,
+                topic=known_args.output,
+                expansion_service=expansion_service,
+            )
+        )
+
+        logging.getLogger().setLevel(logging.DEBUG)
+        logging.info("Building pipeline ...")
+
+
+if __name__ == "__main__":
+    run()
+{{< /highlight >}}
+
+The pipeline script is added to a Python package under a folder named `word_len`. A simple module named `run` is created, because it is executed as a module, for example, `python -m ...`. When I ran the pipeline as a script, I encountered an error. This packaging method is for demonstration only. For a recommended way of packaging a pipeline, see [Managing Python Pipeline Dependencies](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/) .
```

Comment:
```suggestion
The pipeline script is added to a Python package under a folder named `word_len`. A simple module named `run` is created, because it is executed as a module, for example, `python -m ...`. When I ran the pipeline as a script, I encountered an error. This packaging method is for demonstration only. For a recommended way of packaging a pipeline, see [Managing Python Pipeline Dependencies](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/).
```